File: app/src/main/java/com/mlkit/textrecognition/app/di/components/ViewModelModule.java
package com.mlkit.textrecognition.app.di.components;

import com.mlkit.textrecognition.presentation.utils.AppViewModelFactory;

import androidx.lifecycle.ViewModelProvider;
import dagger.Binds;
import dagger.Module;

@Module
public abstract class ViewModelModule {

    @Binds
    abstract ViewModelProvider.Factory bindViewModelFactory(AppViewModelFactory factory);

//    @Binds
//    @IntoMap
//    @ViewModelKey(SplashViewModel.class)
//    abstract ViewModel bindSplashViewModel(SplashViewModel splashViewModel);
}

File: app/src/main/java/com/mlkit/textrecognition/app/di/components/ViewModelModule.kt
package com.mlkit.textrecognition.app.di.components

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.mlkit.textrecognition.app.di.annotation.ViewModelKey
import com.mlkit.textrecognition.presentation.modules.main.MainViewModel
import com.mlkit.textrecognition.presentation.utils.AppViewModelFactory
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap

@Module
abstract class ViewModelModule {

    @Binds
    internal abstract fun bindViewModelFactory(factory: AppViewModelFactory): ViewModelProvider.Factory

    @Binds
    @IntoMap
    @ViewModelKey(MainViewModel::class)
    internal abstract fun bindMainViewModel(mainViewModel: MainViewModel): ViewModel
}